tools/sense_studio/training.py

```diff
@@ -106,20 +113,11 @@ def send_training_logs(msg):
 
     path = project_utils.lookup_project_path(project)
     img_path = os.path.join(path, 'checkpoints', output_folder, 'confusion_matrix.png')
-    if os.path.exists(img_path):
-        img_path = url_for('training_bp.confusion_matrix',
-                           project=msg['project'],
-                           output_folder=msg['outputFolder'])
-        emit('success', {'status': 'Complete', 'img_path': img_path})
+    if confmat_event.is_set() and os.path.exists(img_path):
+        with open(img_path, 'rb') as f:
+            data = f.read()
+        img_base64 = base64.b64encode(data)
+        if img_base64:
```

Comment:
I put this because I did some testing on this. This will cause failure in case the image is being written on drive by process and we cancel it in between which will result in empty text file in place of image. Which on web-page will show broken image icon that will cause problem in user understanding it. So this condition ensures that if the file is written completely or not and if yes, then send it.

Reply:
Thanks for the fix! However, we should also send the `failed` response in this case.